opm/common/data/SimulationDataContainer.hpp

```diff
@@ -86,6 +95,12 @@ namespace Opm {
         const std::vector<double>& facepressure() const;
         const std::vector<double>& faceflux    () const;
 
+
+        std::unordered_map<std::string, std::vector<double>>::iterator celldata_begin();
+        std::unordered_map<std::string, std::vector<double>>::iterator celldata_end();
+        const std::unordered_map<std::string, std::vector<double>>::const_iterator celldata_cbegin() const;
+        const std::unordered_map<std::string, std::vector<double>>::const_iterator celldata_cend() const;
```

Comment:
This should probably be a `const_iterator` rather than a `const const::iterator`.  The latter isn't really very useful because you can't do anything to it--including incrementing.  Also, having a few type aliases in place should make the return type more palatable.  For instance, something like
```C++
using StateVariable  = std::vector<double>;
using Key            = std::string;
using DynamicState   = std::unordered_map<Key, StateVariable>;
using iterator       = DynamicState::iterator;
using const_iterator = DynamicState::const_iterator;
```